File: assignments/solutions/02-first-java/palindrome.java
// Palindrome String

import java.util.*;

public class palindrome {
    public static void main (String args[]) {
        Scanner scan = new Scanner(System.in);
        System.out.println("Palindrome String");
        System.out.print("Enter text: ");
        String input = scan.nextLine();
        String reverse = "";
        for(int i = input.length()-1; i >= 0; i--) {
            reverse += input.charAt(i);
        }
        if(input.toLowerCase().equals(reverse.toLowerCase()))
            System.out.println(input + " is palindrome");
        else
            System.out.println(input + " is not palindrome");
    }
}

/*
Output:

Palindrome String
Enter text: Hello
Hello is not palindrome

Palindrome String
Enter text: MadAm
MadAm is palindrome
*/

File: assignments/solutions/02-first-java/palindrome.js
/**
 * Palindrome String
 * 
 * @param {String} input
 * @returns {boolean}
 */

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const palindrome = (input) => {
    return input.toLowerCase() === input.split('').reverse().join('').toLowerCase();
}

console.log('Palindrome String');
rl.question('Enter text: ', input => {
    console.log(`${input} is ${palindrome(input) ? "palindrome" : "not palindrome"}`);
    rl.close();
});
 
/**
 * Output:
 * 
 * Palindrome String
 * Enter text: mAdaM
 * mAdaM is palindrome
 * 
 * Palindrome String
 * Enter text: Hello
 * Hello is not palindrome
 */